semver.py

```diff
@@ -474,17 +437,18 @@ def compare(self, other):
         0
         """
         cls = type(self)
-        if isinstance(other, string_types):
+        if isinstance(other, STRING_TYPES):
             other = cls.parse(other)
         elif isinstance(other, dict):
             other = cls(**other)
         elif isinstance(other, (tuple, list)):
             other = cls(*other)
         elif not isinstance(other, cls):
             raise TypeError(
-                "Expected str or {} instance, but got {}".format(
-                    cls.__name__, type(other)
-                )
+                (
+                    "Expected str, bytes, dict, tuple, list, or {} instance, "
+                    "but got {}"
+                ).format(cls.__name__, type(other))
```

Comment:
```suggestion
                    f"Expected str, bytes, dict, tuple, list, or {cls.__name__} instance, " \
                    f"but got {type(other)}"
```
Since python 3.6 we can use f strings 🎉